### PR TITLE
Event for withers and ender dragon block destruction

### PR DESCRIPTION
**Pull request by Fadenfire**
Basically an event form of Block#canEntityDestroy. Cancelling the event prevents block damage.

### CODE REVIEW

**Comment by mezz** [comment]
Looks good. 
Please explain what exactly you want to use this new event for.

**Comment by Fadenfire** [comment]
Basically I want to have to effects of mobGriefing applied to the wither and ender dragon, but not everything else like creepers and endermen.

**Comment by mezz** [comment]
Could that be a forge config option then?

**Comment by Fadenfire** [comment]
Probably not, no. I would like to check if the entity is within range of a certain block, if so prevent them from destroying blocks.

**Comment by mezz** [comment]
Ok, that's not really what you said before at all.
Makes sense.

**Comment by Fadenfire** [comment]
Sorry I'm confusing you. I want to use this in two different mods, one where you can enable and disable wither and ender dragon block destruction in the mod's config file and another that adds a block that does what I just said above.

**Comment by mezz** [comment]
It's important to tell us all about the cool things you want to do with the new event so that adding it is justified. It's also necessary so that anyone reviewing can brainstorm to see if there is a simpler way to accomplish what you want to do. Without that, there is no way your PR will be accepted. I want either your problem to be resolved or your PR accepted, so that's why I'm harassing you to explain more. Hope that helps.

**Comment by HenryLoenwind** [comment]
I'm quite sure* you missed the Wither's "blue head" projectile...

(* No workspace access atm)

**Comment by Madsthunder** [comment]
@Silly511 Why not do this for Zombies and Husks also?

**Comment by Fadenfire** [comment]
@HenryLoenwind Yeah I guess I did. I'll fix that.

@Madsthunder How do zombies and husks destroy blocks?

**Comment by HenryLoenwind** [comment]
Zombies break down doors on difficulty hard...

**Comment by Fadenfire** [comment]
Oh sorry, I always though they just opened them. I've never really played awhile on hard. Yeah I guess I could expand this to include zombies.

File: patches/minecraft/net/minecraft/entity/ai/EntityAIBreakDoor.java.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- ../src-base/minecraft/net/minecraft/entity/ai/EntityAIBreakDoor.java
++++ ../src-work/minecraft/net/minecraft/entity/ai/EntityAIBreakDoor.java
+@@ -21,7 +21,7 @@
+         {
+             return false;
+         }
+-        else if (!this.field_75356_a.field_70170_p.func_82736_K().func_82766_b("mobGriefing"))
++        else if (!this.field_75356_a.field_70170_p.func_82736_K().func_82766_b("mobGriefing") || !net.minecraftforge.event.ForgeEventFactory.onEntityDestroyBlock(this.field_75356_a, this.field_179507_b, this.field_75356_a.field_70170_p.func_180495_p(this.field_179507_b)))
```
Comment (LexManos) [inline patches/minecraft/net/minecraft/entity/ai/EntityAIBreakDoor.java.patch]: worth adding a canEntityDestroy check in here?

Comment (Fadenfire) [reply patches/minecraft/net/minecraft/entity/ai/EntityAIBreakDoor.java.patch]: Sure.

**Pull request merged**